wallet-new/docs/updates.md

```diff
@@ -0,0 +1,563 @@
+At this stage, closing Daedalus won't have any effect, as upon the next startup the RESTful call would be
+performed again.
+
+**N.B** A very fair point could be made that we are giving clients too many responsibility and we are offloading
+too much logic to them. If that's the case, we can easily create a new endpoint such as `GET /api/v1/update/severity`
+which would reply with a JSON `{ "updateSeverity": "mandatory" }` in case an update is available, is not fully
+applied and is marked as `Mandatory` (names and payloads might vary to make them more fitting).
+
+-   (1.1) As a Daedalus user, I want to be able to download an update.
+
+After having grabbed a `WalletSoftwareUpdate` via `GET /api/v1/update` Daedalus is now in the position to
+react to user's input. In this story, the user would choose to continue with the update process. To do so, the update
+would need to be downloaded. More specifically, the user would call `POST /api/v1/update/download` which
+would begin the download process. No state change would be necessary in the backend yet. Story `1.2.` assumes
+that this endpoint has been called and it details its behaviour.
+
+-   (1.2) As a Daedalus user, I want to be able to see the download progress for an update.
+
+To accomplish this, one possible idea would be to use [chunked transfer encoding](https://en.wikipedia.org/wiki/Chunked_transfer_encoding)
+to send Daedalus information about the progress of the update. For this to work, we would need to store
+the total size of the binary we need to download somewhere, so that Daedalus could compute a progress
+percentage by doing simple math.
+
+Once the process is complete, the backend would update the `UpdateState` of a `WalletSoftwareUpdate`
+to remember the fact this update was already downloaded, also recording the `FilePath` where this binary lives and
+the `Sha256Checksum` to be used later on in case an update is downloaded but not applied. This will allow the update
+process to skip a new download, whilst ensuring integrity of the data.
+
+-   (2) As a Daedalus user, I want to be able to dismiss an update, unless this is a critical one;
+
+In this scenario, Daedalus would prompt the user that a new update is available (and is non critical) but
+the user would dismiss the update, for example closing a modal in Daedalus. As a consequence, Daedalus
+would call `PUT /api/v1/update/skip` to skip the update. The backend would register this information by
+updating the `UpdateState` to `Skipped`. This endpoint is idempotent, and can be called more than once.
+
+-   (2.1) As a Daedalus user, I want to be sure that once an update is dismissed, it won't be silently applied
+    next time I restart the program;
+
+This will be ensured by design, as it's only when the user decides to apply the update that the installer is
+moved from the download directory into the folder the Cardano Launcher monitors and uses for the update process.
+
```

Comment:
Should we include a _separate_ story, when the user _deliberatly_ wants an update to be applied not immediately, but only after manual restart?

Reply:
Yes, I think we should as it's a standard case. Conceptually, it doesn't change too much; we will still move the downloaded installer in the directory the launcher will inspect, but we won't trigger a shutdown, as that will be left to the user.